packages/certs-v5/test/commands/certs/auto/enable.--wait.js

```diff
@@ -165,4 +174,45 @@ describe('heroku certs:auto:enable --wait', function () {
         acmApi.done()
       })
   })
+
+  it('does not wait if all certs are issued when first checked', () => {
+    const notifySpy = sandbox.spy(require('@heroku-cli/notifications'), 'notify')
+
+    let acmApi = nock('https://api.heroku.com', {
+      reqheaders: { 'Accept': 'application/vnd.heroku+json; version=3.cedar-acm' }
+    })
+    acmApi.post('/apps/example/acm', {}).reply(200, { acm: true })
+
+    let domainsApi = nock('https://api.heroku.com')
+    domainsApi.get('/apps/example/domains').reply(200, [
+      { 'kind': 'heroku', 'hostname': 'tokyo-1050.herokuapp.com', 'cname': null },
+      { 'kind': 'custom', 'hostname': 'heroku-acm.heroku-cli-sni-test.com', 'cname': 'heroku-acm.heroku-cli-sni-test.com.herokudns.com'},
+      { 'kind': 'custom', 'hostname': 'heroku-san-test.heroku-cli-sni-test.com', 'cname': 'heroku-san-test.heroku-cli-sni-test.com.herokudns.com'},
+    ])
+    let now = new Date().toISOString()
+    // Precess finished before the first check, no need to wait
+    domainsApi.get('/apps/example/domains').reply(200, [
+      { 'kind': 'heroku', 'hostname': 'tokyo-1050.herokuapp.com', 'cname': null },
+      { 'kind': 'custom', 'hostname': 'heroku-acm.heroku-cli-sni-test.com', 'cname': 'heroku-acm.heroku-cli-sni-test.com.herokudns.com', 'acm_status': 'cert issued', 'updated_at': now },
+      { 'kind': 'custom', 'hostname': 'heroku-san-test.heroku-cli-sni-test.com', 'cname': 'heroku-san-test.heroku-cli-sni-test.com.herokudns.com', 'acm_status': 'cert issued', 'updated_at': now },
+    ])
+    domainsApi.get('/apps/example/domains').reply(200, [
+      { 'kind': 'heroku', 'hostname': 'tokyo-1050.herokuapp.com', 'cname': null },
+      { 'kind': 'custom', 'hostname': 'heroku-acm.heroku-cli-sni-test.com', 'cname': 'heroku-acm.heroku-cli-sni-test.com.herokudns.com', 'acm_status': 'cert issued', 'updated_at': now },
+      { 'kind': 'custom', 'hostname': 'heroku-san-test.heroku-cli-sni-test.com', 'cname': 'heroku-san-test.heroku-cli-sni-test.com.herokudns.com', 'acm_status': 'cert issued', 'updated_at': now },
+    ])
+
+    return certs.run({
+      app: 'example',
+      args: ['--wait'],
+      flags: {wait: true}
+    })
+      .then(function () {
+        expect(notifySpy.called).to.equal(true)
+        expect(cli.stderr).to.equal('Enabling Automatic Certificate Management... starting.\n')
+        expect(cli.stdout).to.equal('=== Your certificate will now be managed by Heroku.  Check the status by running heroku certs:auto.\n')
+        domainsApi.done()
```

Comment:
Sadly I don't know too much about the **nock** library, but in this case only one of the expectations is meet, the first call, but you are adding two, so in my head this test should be failing. Because the idea of this test is to validate that only one call to /apps/xxx/domains was made, please add an assertion on that (if it is possible)

Reply:
Nock will check if all the declared API calls were made, no need of an extra assertion.

The extra call (on line 199) is needed because the domains are queried one last time to print a summary at the end of the command.

Reply:
Ahhhh, I didn't noticed that last call